File: pkg/model/param.go
package model

// Parameter of an op
type Param struct {
  Dir       *DirParam `yaml:"dir,omitempty"`
  File      *FileParam `yaml:"file,omitempty"`
  NetSocket *NetSocketParam `yaml:"netSocket,omitempty"`
  String    *StringParam `yaml:"string,omitempty"`
}

// Directory parameter of an op
type DirParam struct {
  Description string `yaml:"description"`
  IsSecret    bool `yaml:"isSecret"`
  Name        string `yaml:"name"`
}

// File parameter of an op
type FileParam struct {
  Description string `yaml:"description"`
  IsSecret    bool `yaml:"isSecret"`
  Name        string `yaml:"name"`
}

// Network socket parameter of an op
type NetSocketParam struct {
  Constraints *NetSocketConstraints `yaml:"constraints"`
  Description string `yaml:"description"`
  IsSecret    bool `yaml:"isSecret"`
  Name        string `yaml:"name"`
}

// String parameter of an op
type StringParam struct {
  Constraints *StringConstraints `yaml:"constraints"`
  Default     string `yaml:"default,omitempty"`
  Description string `yaml:"description"`
  IsSecret    bool `yaml:"isSecret"`
  Name        string `yaml:"name"`
}

File: lib/model/param.js
/**
* Typed param of an op
* @typedef {Object} Param
* @property {ArrayParam|null} [array] array typed param
* @property {BooleanParam|null} [boolean] boolean typed param
* @property {DirParam|null} [dir] dir typed param
* @property {FileParam|null} [file] string typed param
* @property {NumberParam|null} [number] number typed param
* @property {ObjectParam|null} [object] object typed param
* @property {SocketParam|null} [socket] socket typed param
* @property {StringParam|null} [string] string typed param
*/

/**
* Param of type object
* @typedef {Object} ArrayParam
* @property {Object} [constraints]
* @property {Array<*>|null} [default] default value for param; if any
* @property {string} [description]
* @property {boolean} [isSecret]
*/

/**
* Param of type boolean
* @typedef {Object} BooleanParam
* @property {boolean|null} [default] default value for param; if any
* @property {string} [description]
*/

/**
* Param of type dir
* @typedef {Object} DirParam
* @property {string|null} [default] default value for param; if any
* @property {string} [description]
* @property {boolean} [isSecret]
*/

/**
* Param of type file
* @typedef {Object} FileParam
* @property {string|null} [default] default value for param; if any
* @property {string} [description]
* @property {boolean} [isSecret]
*/

/**
* Param of type number
* @typedef {Object} NumberParam
* @property {Object} [constraints]
* @property {number|null} [default] default value for param; if any
* @property {string} [description]
* @property {boolean} [isSecret]
*/

/**
* Param of type object
* @typedef {Object} ObjectParam
* @property {Object} [constraints]
* @property {Object|null} [default] default value for param; if any
* @property {string} [description]
* @property {boolean} [isSecret]
*/

/**
* Param of type socket
* @typedef {Object} SocketParam
* @property {string|null} [default] default value for param; if any
* @property {boolean} [isSecret]
*/

/**
* Param of type string
* @typedef {Object} StringParam
* @property {Object} [constraints]
* @property {string|null} [default] default value for param; if any
* @property {string} [description]
* @property {boolean} [isSecret]
*/
